File: Difficulty: Easy/Common in 3 Sorted Arrays/common-in-3-sorted-arrays.cpp
class Solution {
  public:
    vector<int> commonElements(vector<int> &a, vector<int> &b, vector<int> &c) {
        // code here
        int i = 0, j = 0, k = 0;
        int n1 = a.size(), n2 = b.size(), n3 = c.size();
        vector<int> common;
    
        // Traverse all three arrays using three pointers
        while (i < n1 && j < n2 && k < n3){
            
            // If all elements are equal, it's a common element
            if (a[i] == b[j] && b[j] == c[k]){
                common.push_back(a[i]);
                i++;
                j++;
                k++;
    
                // Skip duplicates in all arrays
                while (i < n1 && a[i] == a[i - 1])
                    i++;
                while (j < n2 && b[j] == b[j - 1])
                    j++;
                while (k < n3 && c[k] == c[k - 1])
                    k++;
            }
            
            // Move the pointer with the smallest value
            else if (a[i] < b[j])
                i++;
            else if (b[j] < c[k])
                j++;
            else
                k++;
        }
    
        return common;
    }
};

File: Difficulty: Easy/Common in 3 Sorted Arrays/common-in-3-sorted-arrays.java
class Solution {
    public ArrayList<Integer> commonElements(int[] a, int[] b, int[] c) {
        // code here
        ArrayList<Integer> list=new ArrayList<>();
        int i=0;
        int j=0;
        int k=0;
        
        int l1=a.length;
        int l2=b.length;
        int l3=c.length;
        
        
        while(i<l1 && j<l2 && k<l3)
        {
            if(a[i]==b[j] && b[j]==c[k])
            {
                if(list.size()==0)list.add(a[i]);
                else if(list.get(list.size()-1) != a[i])list.add(a[i]);
                i++;
                j++;
                k++;
            }
            else if(a[i]<b[j] || a[i]<c[k])i++;
            else if(b[j]<c[k] || b[j]<a[i])j++;
            else if(c[k]<b[j] || c[k]<a[i])k++;
        }
        return list;
    }
}